### PR TITLE
RGRIDT-770: Check if column exists before adding them again

### PR DESCRIPTION
**Pull request by glundgren93**
### What was happening
* Whenever we used server-side search, the columns were being duplicated

### What was done
* Added a condition before the step of adding columns

### Test Steps
1. Open sample
2. On server-side search, search for pink
3. Check if new columns were not added. 

### Screenshots
(prefer animated gif)

### Checklist
* [ ] tested locally
* [ ] documented the code
* [ ] clean all warnings and errors of eslint
* [ ] requires changes in OutSystems (if so, provide a module with changes)
* [ ] requires new sample page in OutSystems (if so, provide a module with changes)

### CODE REVIEW

File: code/src/OSFramework/Grid/AbstractGrid.ts [hunk 1]
```diff
@@ -249,7 +249,9 @@ namespace OSFramework.Grid {
                     this.dataSource.getMetadata(),
                     this.config.allowEdit
                 );
-                generated.forEach((p) => this.addColumn(p));
+                if (this._columns.size === 0) {
+                    generated.forEach((p) => this.addColumn(p));
+                }
```
Comment (rugoncalves) [inline code/src/OSFramework/Grid/AbstractGrid.ts:254]: Does this cover the scenario, when a new column is added as well?

Comment (bmartinho) [reply code/src/OSFramework/Grid/AbstractGrid.ts:254]: I was checking the same and this code only runs in auto generated columns that have metadata (where created with ArrangeData), so no problem with add/remove columns

Comment (rugoncalves) [reply code/src/OSFramework/Grid/AbstractGrid.ts:254]: As discussed, we will leave for now as it is. @glundgren93 if you can add just an improvement to backlog, to (maybe) support swapping data sources in the future.

**Review by rugoncalves — APPROVED** [review]

File: code/src/OSFramework/Grid/AbstractGrid.ts (hunk 1 above)
Comment (bmartinho) [reply code/src/OSFramework/Grid/AbstractGrid.ts:254]: yes!

**Review by bmartinho — APPROVED** [review]
Let's go!

**Review by tmbp95 — APPROVED** [review]

**Pull request merged**